crates/bevy_ecs/src/schedule_v3/graph_utils.rs

```diff
@@ -231,3 +231,131 @@ where
         transitive_closure,
     }
 }
+
+/// Returns the simple cycles in the strongly-connected components of a directed graph.
+///
+/// The algorithm implemented comes from
+/// ["Finding all the elementary circuits of a directed graph"][1] by D. B. Johnson.
+///
+/// [1]: https://doi.org/10.1137/0204007
+pub(crate) fn simple_cycles_in_components<N>(
+    graph: &DiGraphMap<N, ()>,
+    sccs: &[Vec<N>],
+) -> Vec<Vec<N>>
+where
+    N: NodeTrait + Debug,
+{
+    let mut cycles = Vec::new();
+
+    // reporting self-loops as cycles was not part of the original algorithm
+    // for node in graph.nodes() {
```

Comment:
We probably want this, right? We shouldn't be leaving around commented out code regardless though.

Reply:
Sure, I just commented it out because I was trying to get it to work and loops cannot exist in a schedule, `add_system` and `configure_set` error.